### PR TITLE
Lazy gvar

### PR DESCRIPTION
**Pull request by behdad**
Here's a possible sketch of lazy loading of gvar. Two tests fail, and has its limitations.

### CODE REVIEW

**Comment by behdad** [comment]
The main problem is when someone iterates over the .variations they don't get the actual value. Only if they use attr lookup syntax...

**Comment by behdad** [comment]
I fixed `.items()` and `.values()`. The remaining failure is for when the `.variations` dictionary is tested for equality to another dictionary. That's harder to fix.

**Comment by behdad** [comment]
Fixed remaining test.

File: Lib/fontTools/ttLib/tables/_g_v_a_r.py [hunk 1]
```diff
@@ -97,23 +137,17 @@ def decompile(self, data, ttFont):
...
-			try:
-				self.variations[glyphName] = decompileGlyph_(
-					numPointsInGlyph, sharedCoords, axisTags, gvarData)
-			except Exception:
-				log.error(
-					"Failed to decompile deltas for glyph '%s' (%d points)",
-					glyphName, numPointsInGlyph,
-				)
-				raise
+			self.variations[glyphName] = get_lambda(gvarData, numPointsInGlyph)
```
Comment (justvanrossum) [inline Lib/fontTools/ttLib/tables/_g_v_a_r.py:150]: Wouldn't this be prettier with a `functools.partial`?

**Review by justvanrossum — APPROVED** [review]
Awesome! LGTM.

File: Lib/fontTools/ttLib/tables/_g_v_a_r.py (hunk 1 above)
Comment (behdad) [reply Lib/fontTools/ttLib/tables/_g_v_a_r.py]: Thanks. Fixed.

Comment (justvanrossum) [reply Lib/fontTools/ttLib/tables/_g_v_a_r.py]: I was rather thinking that you could partial `decompileGlyph_` directly. I don't think `func` is really needed at all?

Comment (behdad) [reply Lib/fontTools/ttLib/tables/_g_v_a_r.py]: Oh, you're right! Let me do that. Thanks.

Comment (behdad) [reply Lib/fontTools/ttLib/tables/_g_v_a_r.py]: Done. So much nicer. Thanks.
https://github.com/fonttools/fonttools/pull/2738 is also ready :D.

**Pull request merged**